common/lib/xmodule/xmodule/discussion_module.py

```diff
@@ -42,7 +44,20 @@ class DiscussionFields(object):
     sort_key = String(scope=Scope.settings)
 
 
+def has_permission(user, permission, course_id):
+    """
+    Copied from django_comment_client/permissions.py because I can't import
+    that file from here. It causes the xmodule_assets command to fail.
+    """
+    return any(role.has_permission(permission)
+               for role in user.roles.filter(course_id=course_id))
+
+
+@XBlock.wants('user')
```

Comment:
This seems like a bit of an in-between case. It's possible to define sane behavior in the absence of a user, but we (currently) expect that there will be a user when rendering a discussion module. It's a little sad to me to implement code that we expect to never run in production, but I can live with it.